src/syntaxtree.c

```diff
@@ -0,0 +1,898 @@
+//
+//  syntaxtree.c
+//
+#include "syntaxtree.h"
+
+#include "bool.h"
+#include "calls.h"
+#include "code.h"
+#include "compiled.h" // C_MAKE_INTEGER_BAG
+#include "exprs.h"
+#include "gvars.h"
+#include "error.h"
+#include "lists.h"
+#include "modules.h"
+#include "plist.h"
+#include "records.h"
+#include "stats.h"
+#include "stringobj.h"
+#include "precord.h"
+#include "system.h"
+#include "vars.h"
+
+#include <stdarg.h>
+#include <ctype.h>
+
+typedef UInt4 LVar;
+typedef UInt4 HVar;
+typedef UInt  GVar;
+
+typedef Obj (*CompileFuncT)(Obj node, Expr expr);
+typedef Obj (*CompileArgT)(Expr expr);
+
+typedef struct {
+    const Char * argname;
+    CompileArgT  argcomp;
+} ArgT;
+
+typedef struct {
+    UInt         tnum;
+    CompileFuncT compile;
+    const Char * name;
+    UInt         arity;
+    ArgT         args[6]; /* This is not really a restriction */
```

Comment:
I don't quite understand this comment. Do you mean to express that you actually allow instances to have more than 6 entries here? But then why not use a flexible array member, i.e. `ArgT args[];`, then this is self-explanatory and unambiguous.

Reply:
The storage size of args has to be known at compile time if I want the compilers to be statically setup (otherwise I get `error: initialization of flexible array member in a nested context`).

The arity of compilers has a statically known upper bound though (it's currently 4 if I am not mistaken).

I admit that the comment should be clearer though.